use-package.el

```diff
@@ -656,47 +656,40 @@ manually updated package."
 ;; :bind, :bind*
 ;;
 
-(defsubst use-package-is-sympair (x &optional allow-vector)
-  "Return t if X has the type (STRING . SYMBOL)."
+(defsubst use-package-is-pair (x car-pred cdr-pred)
```

Comment:
This will affect all the other keywords that expect pairs too.  Do you have tests for all of them?